### PR TITLE
Integrate checkout with backend

### PR DESCRIPTION
**Pull request by whenzei**

### CODE REVIEW

File: src/components/Customer/CartDialog.vue [hunk 1]
```diff
@@ -212,25 +229,31 @@ export default {
...
-      if (res.status == 200 || res.status == 304) {
+      if (res.status == 200) {
         this.recentAddresses = res.data;
       }
     },
     async populatePromos() {
       const res = await axios.get(`/customer/promos/${this.rid}`);
-      if (res.status == 200 || res.status == 304) {
+      if (res.status == 200) {
+        this.promos.push("None");
```
Comment (jelneo) [inline src/components/Customer/CartDialog.vue:245]: Shouldn't this option still be there if fetching of promos fail?

File: src/components/Customer/CartDialog.vue [hunk 2]
```diff
@@ -239,22 +262,67 @@ export default {
...
-    checkout() {
-      const selectedPromo = this.promosMap[this.promo].pid;
-      console.log(selectedPromo);
+    async checkout() {
+      this.submitted = true;
+      this.dialog = false;
+      let orderDetails = this.preprocessOrder();
+      try {
+        const res = await axios.post("/customer/checkout", orderDetails);
+        await this.wait(1000);
```
Comment (jelneo) [inline src/components/Customer/CartDialog.vue:271]: why is a delay introduced?

File: src/components/Customer/CartDialog.vue (hunk 1 above)
Comment (whenzei) [reply src/components/Customer/CartDialog.vue:245]: It should not populate if it fails.

File: src/components/Customer/CartDialog.vue (hunk 2 above)
Comment (whenzei) [reply src/components/Customer/CartDialog.vue:271]: So that my transition is visible

**Pull request merged**